Labib/org panel UI

Kashshaf-Labib opened the pull request.

Kashshaf-Labib committed:
feat(SignIn): refactor sign-in logic into separate component and remove unused code

Kashshaf-Labib committed:
fix(AuthProvider): fix AuthProvider and modularize code

Kashshaf-Labib committed:
feat(ViewOrganization): enhance image upload and preview functionality, refactor organization details layout

Kashshaf-Labib committed:
feat(ViewTurfs): implement turf viewing page with filtering and search functionality, add TurfGrid and TurfCard components and add fetchTurfsByOrganization API for retrieving turf data

Kashshaf-Labib committed:
feat(TurfDetails): add turf detail page and update routing, enhance TurfCard link structure, and implement fetchTurfById API

Kashshaf-Labib committed:
feat (Turf Update): add api and frontend implementation for turf update.

Kashshaf-Labib committed:
fix(TurfDetailPage component): Update ApiResponse interface to ensure consistent formatting and improve ui

Kashshaf-Labib committed:
feat(TimeSlot API): add timeslot api and interfaces

Kashshaf-Labib committed:
feat(TimeSlot ui): implement frontend for generating timeslots for turfs.

Kashshaf-Labib committed:
fix(Timeslot error handling): improve error message for timeslot generation failure

Kashshaf-Labib committed:
feat(booking):add booking service backend

Kashshaf-Labib committed:
fix(auth): update access check to use hasAccess instead of hasRole

Kashshaf-Labib committed:
fix(auth.ts): fix interface

Kashshaf-Labib committed:
Merge branch 'feature/org-panel-ui' of https://github.com/Learnathon-By-Geeky-Solutions/devorbit into labib/org-panel-ui

Kashshaf-Labib committed:
Merge branch 'dev' of https://github.com/Learnathon-By-Geeky-Solutions/devorbit into labib/org-panel-ui

Kashshaf-Labib committed:
Implement feature X to enhance user experience and optimize performance

Warhammer4000 commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/warhammer4000/pr-checks/e022f27d-578b-4609-aaa0-220407a78bb7?refs=4b777b666828626d368cf1db83c69553067939cc&source=prChecksComment)

Kashshaf-Labib committed:
feat(auth): implement authentication context and user management functions

feat(api): add fetchAvailableTimeSlots and getUserOrganizations APIs

fix(routes): rename complete booking route to complete-stripe

chore(deps): add js-cookie and its types for token management

refactor(user): remove unused hasOrganizationRole method

style: clean up layout files and remove commented code

chore(tsconfig): add tsconfig for organization module

type: define ApiError and IUser interfaces

Kashshaf-Labib committed:
feat(Booking Initial Phase): start Booking in Client app
